ledger/sandbox/src/main/scala/com/digitalasset/platform/sandbox/SandboxServer.scala

```diff
@@ -73,14 +73,31 @@ object SandboxServer {
 
   // if requested, initialize the ledger state with the given scenario
   private def createInitialState(config: SandboxConfig, context: SandboxContext)
-    : (ActiveContractsInMemory, ImmArray[LedgerEntryWithLedgerEndIncrement], Option[Instant]) =
+    : (ActiveContractsInMemory, ImmArray[LedgerEntryWithLedgerEndIncrement], Option[Instant]) = {
+    // [[ScenarioLoader]] needs all the packages to be already compiled --
+    // make sure that that's the case
+    if (config.eagerPackageLoading || config.scenario.nonEmpty) {
+      for (pkg <- context.packageContainer.packages) {
```

Comment:
Minor: since you are using `pkg` as a tuple, it would probably make it more readable if you could deconstruct it here to give more meaningful names than `pkg._1` and `pkg._2`.

Reply:
done